sphinx/themes/basic/static/searchtools.js

```diff
@@ -8,522 +8,493 @@
  * :license: BSD, see LICENSE for details.
  *
  */
+"use strict";
 
-if (!Scorer) {
-  /**
-   * Simple result scoring code.
-   */
-  var Scorer = {
-    // Implement the following function to further tweak the score for each result
-    // The function takes a result array [filename, title, anchor, descr, score]
-    // and returns the new score.
-    /*
-    score: function(result) {
-      return result[4];
-    },
-    */
-
-    // query matches the full name of an object
-    objNameMatch: 11,
-    // or matches in the last dotted part of the object name
-    objPartialMatch: 6,
-    // Additive scores depending on the priority of the object
-    objPrio: {0:  15,   // used to be importantResults
-              1:  5,   // used to be objectResults
-              2: -5},  // used to be unimportantResults
-    //  Used when the priority is not in the mapping.
-    objPrioDefault: 0,
-
-    // query found in title
-    title: 15,
-    partialTitle: 7,
-    // query found in terms
-    term: 5,
-    partialTerm: 2
-  };
-}
-
-if (!splitQuery) {
-  function splitQuery(query) {
-    return query.split(/\s+/);
+/**
+ * Simple result scoring code.
+ */
+const Scorer = {
```

Comment:
There's a behavior change here -- this should stay a `var`, that only gets assigned under the `if (!Scorer) {...}` block.

Otherwise, someone can't override the scorer settings; which is what this was intended for.

Reply:
Ahh, I wondered why it was -- i was under the impression it was some ancient workaround (most likely for IE) that never got removed. Will revert, and add a note speaking as to why it is defined as such.